### ISSUE TITLE
Vertical resizer for Karpenter

### ISSUE THREAD

**Issue by easildur24** [aws/karpenter#3194]
### Tell us about your request

My team just started using Karpenter in our clusters. One thing that was missing from the Karpenter deployment is the nanny that we used to have with Cluster Autoscaler. This is a bit inconvenient given that our clusters grow in time as more services are deployed on the clusters and we have to monitor the memory usage and manually bump up the resource requests. Do we already have something that we can use for Karpenter or is it on the roadmap?

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

Stated in the request.

### Are you currently working around this issue?

Manually bumping up the resources requests for now.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by ellistarn** [comment]
How does this work for CAS?

**Comment by easildur24** [comment]
> How does this work for CAS?

This is what we use: https://github.com/kubernetes/autoscaler/blob/master/addon-resizer/README.md

**Comment by ellistarn** [comment]
Have you tried using this with Karpenter? I'm not aware of anyone trying this yet.

**Comment by easildur24** [comment]
Yeah I can give it a try. Was just wondering if there's a custom one made for Karpenter.

**Comment by ellistarn** [comment]
Not yet, but looking forward to what you learn.

**Comment by easildur24** [comment]
Ellis, while I have you on the thread, could you give a little insight on what causes Karpenter to use more CPU and memory? Do they grow proportionally with the number of the nodes and pods in the cluster?

**Comment by ellistarn** [comment]
#pods and #nodes definitely contribute. Consolidation definitely adds to it. We haven't profiled a ton, but you can enable a flag to turn on profiling https://karpenter.sh/v0.22.0/concepts/settings/

**Comment by ellistarn** [comment]
Wow thanks for the report -- we'll dig into this.

**Comment by ellistarn** [comment]
Do you mind cutting a new issue detailing your observations?

**Comment by vassilvk** [comment]
Done: #3209

**Comment by sftim** [comment]
There is also https://github.com/kubernetes-sigs/cluster-proportional-autoscaler which, I think, might help.
